### PR TITLE
src/mount: kernel mount command returning misleading error message

### PR DESCRIPTION
**Pull request by neesingh-rh**
Kernel Mount command returning error as MDS is laggy for unauthorized client,
correct the debug log.

Fixes: https://tracker.ceph.com/issues/63866
Signed-off-by: Neeraj Pratap Singh <neesingh@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/mount/mount.ceph.c [hunk 1]
```diff
@@ -912,7 +912,8 @@ static int do_mount(const char *dev, const char *node,
 			fprintf(stderr, "mount error: ceph filesystem not supported by the system\n");
 			break;
 		case EHOSTUNREACH:
-			fprintf(stderr, "mount error: no mds server is up or the cluster is laggy\n");
+			fprintf(stderr, "mount error: no mds server is available — "
+			"it may be laggy or down, or you may not be authorized\n");
```
[the remark this answers is not in the document]
Comment (neesingh-rh) [reply src/mount/mount.ceph.c]: Updated it with better message

File: src/mount/mount.ceph.c [hunk 2]
```diff
@@ -912,7 +912,8 @@ static int do_mount(const char *dev, const char *node,
 			fprintf(stderr, "mount error: ceph filesystem not supported by the system\n");
 			break;
 		case EHOSTUNREACH:
-			fprintf(stderr, "mount error: no mds server is up or the cluster is laggy\n");
+			fprintf(stderr, "mount error: no mds server is up — "
+			"the cluster may be laggy, or you may not be authorized\n");
```
Comment (dparmar18) [inline src/mount/mount.ceph.c:916]:
```suggestion
			fprintf(stderr, "mount error: no MDS(Metadata Server) is up.  "
			"The cluster might be laggy or you may lack authorization \n");
```

Comment (dparmar18) [reply src/mount/mount.ceph.c:916]: the reason for first line - mds server translates to "Metadata Server server" while sounds a bit weird to me

**Review by dparmar18 — APPROVED** [review]
LGTM

**Comment by neesingh-rh** [comment]
@vshankar PTAL, Does it also need testing?

**Comment by vshankar** [comment]
> @vshankar PTAL, Does it also need testing?

Not really. Will merge it tomorrow.

File: src/mount/mount.ceph.c [hunk 3]
```diff
@@ -912,7 +912,8 @@ static int do_mount(const char *dev, const char *node,
 			fprintf(stderr, "mount error: ceph filesystem not supported by the system\n");
 			break;
 		case EHOSTUNREACH:
-			fprintf(stderr, "mount error: no mds server is up or the cluster is laggy\n");
+			fprintf(stderr, "mount error: no mds (Metadata Server) is up. "
+			"The cluster might be laggy, or you may lack authorization\n");
```
Comment (vshankar) [inline src/mount/mount.ceph.c:916]: nit: "... or you may not be authorized."

**Review by vshankar — APPROVED** [review]

File: src/mount/mount.ceph.c (hunk 3 above)
Comment (vshankar) [reply src/mount/mount.ceph.c:916]: oh, the pr isn't updated with the change.

Comment (neesingh-rh) [reply src/mount/mount.ceph.c:916]: Actually, earlier it was `...or you may not be authorized` but after dhairya's suggestion  changed to `..may lack authorization`. I am on the fence here. If you really want, I will update:)

Comment (neesingh-rh) [reply src/mount/mount.ceph.c]: @vshankar updated with your suggestion.

**Pull request merged**